File: api/proxy.py
import httpx
from fastapi import FastAPI, Request
from fastapi.responses import Response

app = FastAPI()

TARGET_URL = "https://example.com"  # Replace with your backend target

@app.api_route("/{path:path}", methods=["GET", "POST", "PUT", "DELETE", "PATCH", "OPTIONS"])
async def proxy(request: Request, path: str):
    # Construct target URL
    url = f"{TARGET_URL}/{path}"

    # Forward headers and body
    headers = dict(request.headers)
    body = await request.body()

    async with httpx.AsyncClient() as client:
        resp = await client.request(
            method=request.method,
            url=url,
            headers=headers,
            content=body
        )

    return Response(
        content=resp.content,
        status_code=resp.status_code,
        headers=dict(resp.headers)
    )

File: api/proxy.ts
import type { VercelRequest, VercelResponse } from '@vercel/node';
import fetch from 'node-fetch';

const TARGET_URL = 'https://example.com'; // Replace with your backend target

export default async function handler(req: VercelRequest, res: VercelResponse) {
  const path = req.url?.replace(/^\/api\/proxy/, '') || '';
  const url = `${TARGET_URL}${path}`;

  try {
    const response = await fetch(url, {
      method: req.method,
      headers: req.headers as any,
      body: req.method !== 'GET' && req.method !== 'HEAD' ? req.body : undefined,
    });

    const buffer = await response.buffer();

    res.status(response.status);
    response.headers.forEach((value, key) => {
      res.setHeader(key, value);
    });
    res.send(buffer);
  } catch (err: any) {
    res.status(500).json({ error: 'Proxy error', details: err.message });
  }
}
